### DISCUSSION TITLE
How to convert a Kratos session into a jwt token oathkeeper

### DISCUSSION THREAD

**Discussion by stilet** [developer-friendly/ory#2] [Q&A] [answered]
Greetings! Tell me - when will there be an article on how to convert a Kratos session into a jwt token oathkeeper?

**Comment by meysam81** [comment, marked as answer]
@stilet 
Greetings and thanks for showing interest.
If all goes well, I plan to have it out in less than a month, considering all the blog post topics I have in mind in the short term.🙌

**Reply by meysam81** [reply]
@stilet 

The Ory Oathkeeper blog post has been published.

<https://developer-friendly.blog/2024/06/10/ory-oathkeeper-identity-and-access-proxy-server/>

After pondering your question, I can probably offer two approaches to your requirement (if you haven't found one already):
1. Either use the native API **if your client is not browser**: <https://www.ory.sh/docs/kratos/reference/api#tag/frontend/operation/createNativeLoginFlow>
2. Set up the Ory Oathkeeper to use Ory Kratos as an identity provider, and you won't ever need a JWT token for authentication purposes. Here's the corresponding section of the same article above: <https://developer-friendly.blog/2024/06/10/ory-oathkeeper-identity-and-access-proxy-server/#play-2-authenticated-by-ory-kratos>

Please do let me know if this is what you are looking for.

Regards. :raised_hands: 

P.S. I and other members of the community will be more than happy to have you on the [Ory Slack](https://slack.ory.sh/) channel (if you're not already present).

Also, if you're interested, I'd be delighted to have you join the [Developer Friendly Slack](https://communityinviter.com/apps/developerfriendly/join-our-slack) channel.
